libs/esp32boot/esp32init.erl

```diff
@@ -0,0 +1,394 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2023 Davide Bettio <davide@uninstall.it>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(esp32init).
+
+-export([
+    start/0,
+    start_repl/1,
+    start_network/0,
+    maybe_start_network/0,
+    handle_req/3,
+    erase_net_config/0,
+    save_net_config/2
+]).
+
+-record(nc_state, {socket, pending_pid, pending_ref}).
+
+-define(DEFAULT_AP_SSID, <<"AtomVM-ESP32">>).
+-define(DEFAULT_AP_PSK, <<"esp32default">>).
+-define(DEFAULT_CONSOLE_PORT, 2323).
+-define(DEFAULT_WEB_SERVER_PORT, 8080).
+
+start() ->
+    erlang:display("AtomVM init."),
+
+    io:format("Starting application...~n"),
+
+    Exit =
+        try boot() of
+            Result -> {exit, Result}
+        catch
+            Error -> {crash, Error}
+        end,
+    erlang:display(Exit),
+
+    io:format("Looping...~n"),
+
+    loop().
+
+loop() ->
+    receive
+        Msg ->
+            erlang:display({received_message, Msg}),
+            loop()
+    end.
+
+maybe_start_dev_mode(SystemStatus) ->
+    case {SystemStatus, esp:nvs_get_binary(atomvm, dev_mode)} of
+        {_, <<"always">>} ->
+            start_dev_mode();
+        {_, <<"never">>} ->
+            not_started;
+        {ok, undefined} ->
+            not_started;
+        {failed_app_start, undefined} ->
+            start_dev_mode()
+    end.
+
+start_dev_mode() ->
+    avm_pubsub:start(default_pubsub),
+    spawn(?MODULE, maybe_start_network, []),
```

Comment:
Why not use `fun maybe_start_network`?  You don't have to export the function if you do that, and you get the benefit of it not being an anonymous function, so it shows up cleanly in stack traces.

Reply:
oh, right, going to fix this.